### PR TITLE
SALTO-937: Add error in the case of no existing adapter when calling getAdapters…

### PR DESCRIPTION
**Pull request by adamro**
…CredentialsTypes + change the service command flow accordingly

### CODE REVIEW

File: packages/core/src/core/adapters/adapters.ts [hunk 1]
```diff
@@ -14,18 +14,26 @@
...
   names?: ReadonlyArray<string>
 ): Record<string, ObjectType> => {
-  const relevantAdapterCreators = _.isUndefined(names)
-    ? adapterCreators
-    : _.pick(adapterCreators, names)
+  let relevantAdapterCreators: Record<string, Adapter>
+  if (names === undefined) {
+    relevantAdapterCreators = adapterCreators
+  } else {
+    const [existingAdapters, nonExistingAdapters] = _.partition(
```
Comment (omrilit) [inline packages/core/src/core/adapters/adapters.ts:28]: I think that there is no reason to create another variables here using `_.partition()` as you don't really use `existingAdapters` since it equals to `names` in case there was no errors.
```suggestion
  const nonExistingAdapters = names.filter(name => !Object.keys(adapterCreators).includes(name))
```

File: packages/core/src/core/adapters/adapters.ts [hunk 2]
```diff
@@ -14,18 +14,26 @@
...
 import { createDefaultInstanceFromType } from '@salto-io/adapter-utils'
 import adapterCreators from './creators'
 
 export const getAdaptersCredentialsTypes = (
   names?: ReadonlyArray<string>
 ): Record<string, ObjectType> => {
-  const relevantAdapterCreators = _.isUndefined(names)
-    ? adapterCreators
-    : _.pick(adapterCreators, names)
+  let relevantAdapterCreators: Record<string, Adapter>
```
Comment (omrilit) [inline packages/core/src/core/adapters/adapters.ts:24]: nit: Maybe you can extract to `getRelevantAdapterCreators()` so you'll avoid the `let`

File: packages/cli/src/commands/service.ts [hunk 3]
```diff
@@ -74,18 +74,18 @@ const addService = async (
     return CliExitCode.UserInputError
   }
 
-  try {
-    if (!nologin) {
-      const adapterCredentialsType = getAdaptersCredentialsTypes([serviceName])[serviceName]
+  if (!nologin) {
+    const adapterCredentialsType = getAdaptersCredentialsTypes([serviceName])[serviceName]
```
Comment (omrilit) [inline packages/cli/src/commands/service.ts:78]: Why did you move it out of the try block?

Comment (adamro) [reply packages/cli/src/commands/service.ts:78]: Cause this catch block handles specifically login errors with special formatting which is uneeded for this (and other potential) errors

**Review by omrilit — APPROVED** [review]
